rpcs3/Emu/RSX/Common/simple_array.hpp

```diff
@@ -24,11 +24,18 @@ namespace rsx
 		template <size_t Align>
 		void* realloc(void* prev_ptr, [[maybe_unused]] size_t prev_size, size_t new_size)
 		{
+			ensure(reinterpret_cast<usz>(prev_ptr) % Align == 0,
+       			"Pointer not aligned to Align");
+			if (prev_size >= ((new_size + Align - 1) & (0 - Align))) 
```

Comment:
This check is straight-up wrong btw. There is no such thing as forward-aligning the size to guess how long it should be. That's implementation-defined behavior.
However, the free statement below is too important, so I'll merge this as-is and then follow up with a proper check here as well as the unit tests which are missing from this submission.